### PR TITLE
feat(data_classes): API Gateway V2 IAM and Lambda

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

## Description of changes:

Update to support IAM and Lambda authorizer for API
Gateway V2 trigger events

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py [hunk 1]
```diff
@@ -267,6 +267,52 @@ def user_agent(self) -> str:
...
+        """The AWS account ID associated with the request."""
+        return self.get("accountId")
+
+    @property
+    def caller_id(self) -> Optional[str]:
+        """The principal identifier of the caller making the request."""
+        return self.get("callerId")
+
+    @property
+    def cognito_amr(self) -> Optional[List[str]]:
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py:286]: could you add a docstring here?

File: aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py [hunk 2]
```diff
@@ -267,6 +267,52 @@ def user_agent(self) -> str:
         return self["requestContext"]["http"]["userAgent"]
 
 
+class RequestContextV2AuthorizerIam(DictWrapper):
+    @property
+    def access_key(self) -> Optional[str]:
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py:272]: could you add a docstring here?

File: aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py [hunk 3]
```diff
@@ -267,6 +267,52 @@ def user_agent(self) -> str:
...
+        """The AWS organization ID."""
+        return self.get("principalOrgId")
+
+    @property
+    def user_arn(self) -> Optional[str]:
+        """The Amazon Resource Name (ARN) of the effective user identified after authentication."""
+        return self.get("userArn")
+
+    @property
+    def user_id(self) -> Optional[str]:
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py:312]: could you add a docstring here?

File: aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py [hunk 4]
```diff
@@ -276,6 +322,15 @@ def jwt_claim(self) -> Dict[str, Any]:
     def jwt_scopes(self) -> List[str]:
         return self["jwt"]["scopes"]
 
+    @property
+    def get_lambda(self) -> Optional[Dict[str, Any]]:
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py:326]: could you add a docstring here?

File: aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py [hunk 5]
```diff
@@ -276,6 +322,15 @@ def jwt_claim(self) -> Dict[str, Any]:
...
+        return self.get("lambda")
+
+    @property
+    def iam(self) -> Optional[RequestContextV2AuthorizerIam]:
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py:330]: could you add a docstring here?

**Review by heitorlessa — CHANGES_REQUESTED** [review]
LGTM except missing docstrings in certain properties for consistency

**Comment by michaelbrewer** [comment]
@heitorlessa - made those changes. I did have to did around into the cognito docs for some of them and parts of the IAM / Lambda authorizers docs

**Review by heitorlessa — APPROVED** [review]

**Comment by heitorlessa** [comment]
Looks great, thanks for those again @michaelbrewer !!

**Pull request merged**